teamworkApp/lib/importData.py

```diff
@@ -1,26 +1,234 @@
+# muddersOnRails()
+# Maeve Murphy October 12, 2017
+# Last updated: 10-27-2017
+
+# imports data from the .csv file holding responses from the teamwork survey
+
+# resources:
+# https://www.tutorialspoint.com/sqlite/sqlite_select_query.htm
+# https://docs.python.org/2/library/sqlite3.html
+
 import sqlite3
+import csv
+from datetime import datetime, date, time
 from enum import IntEnum
 
 DB = 'db/development.sqlite3'
-
-Style = IntEnum('Style', 'Contributor, Collaborator, Communicator, Challenger', start=0)
 Answer_Value = IntEnum(
-	'Answer_Value', 
+	'Answer_Value',
 	'1234 1243 1324 1342 1423 1432 2134 2143 2314 2341 2413 2431 3124 3142 3214 3241 3412 3421 4123 4132 4213 4231 4312 4321',
 	start=0
 )
 
-def main():
+def getAnswer(answerNum):
+	for ans in range(0, 23):
+		if Answer_Value(ans).name == answerNum:
+			return ans
+	raise ValueError("Invalid response")
+
+def processAnswerData(csvFileName):
+	"""Given the Teamwork Survey csv file, processAnswerData will extract the
+	pertinent information and populate a student dictionary with it.
+	This processing requires very specifically formatted .csv files and it
+	assumes they come from the results of the Teamwork Survey sent out by
+	muddersOnRails.
+	input csvFilename: name (and possibly path to) the csv file corresponding
+		to the data received by respondants to the Teamwork Survey
+	output studentDict: a dictionary. The keys are usernames of students
+		collected from the Teamwork Survey. The value for each key is a list of
+		three pieces of information:
+			[timestamp, expandedResponses, enummedResponses]
+			timestamp: the time at which the student filled out the Teamwork
+				Survey.
+			expandedResponses: the responses to each question expanded out into
+				a string of the four-digit integer corresponding to their
+				rankings, this is a list of 18 responses.
+			enummedResponses: the list of responses to each question enumerated
+				according to the IntEnum Answer_Value. This is a list of 18
+				integers corresponding to the IntEnum."""
+	with open(csvFileName) as csvfile:
+
+		timestamps        = []
+		expandedResponses = []
+		studentDict       = {}
+
+		answerReader = csv.reader(csvfile)
+		studentData  = list(answerReader)
+		# get rid of the row with questions
+		studentData  = studentData[1:]
+
+		for student in studentData:
+			# these are the 18x4 responses to the survey
+			teamworkResponses = student[1:73]
```

Comment:
I think that the 73 should be pulled out as a variable so that it is clearer and easier to find and change later if needed.

Reply:
Especially because you use 72 again in a few lines.

Reply:
on it!